pkg/utils/image.go

```diff
@@ -56,3 +56,12 @@ func chooseImage(key string, specImage *string, iv imagevector.ImageVector) (*st
 	}
 	return pointer.String(ivImage[key].String()), nil
 }
+
+// GetBackupRestoreImage returns the image for `backup-restore` from the given image vector.
+func GetBackupRestoreImage(iv imagevector.ImageVector) (string, error) {
```

Comment:
There is an already existing method `GetEtcdImages` which calls `chooseImage`. If you see its implementation it first chooses the image from `etcd.Spec.Backup.Image` and if that is not there then it will take it from image vector. Instead of introducing a new method which duplicates some code (currently incorrectly), you create individual methods to get etcd and backup-restore images and then call these methods also from `GetEtcdImages`

```golang
func GetEtcdImage(etcd *druidv1alpha1.Etcd, iv imagevector.ImageVector) (string, error) {
  // TODO
}
func GetEtcdBackupRestoreImage(etcd *druidv1alpha1.Etcd, iv imagevector.ImageVector) (string, error) {
  // TODO
}
func GetEtcdImages(etcd *druidv1alpha1.Etcd, iv imagevector.ImageVector) (string, string, error) {
   etcdImg, err := GetEtcdImage(etcd, iv)
   if err != nil {
       return nil, nil, err
   }
   etcdBackupRestoreImg, err := GetEtcdBackupRestoreImage(etcd, iv)
   if err != nil {
       return nil, nil, err
   }
   return etcdImg, etcdBackupRestoreImg, nil
}
```